### PR TITLE
set ClusterOperator status

### PR DESCRIPTION
**Pull request by gabemontero**
replacement for https://github.com/openshift/cluster-samples-operator/pull/33

@bparees ptal

### CODE REVIEW

**Comment by bparees** [comment]
looks good, i'm guessing you haven't had a chance to actually test this (you'll have to create/install the ClusterOperator CRD yourself unless you test it in a real install).

**Comment by gabemontero** [comment]
@bparees pushed an update (squashed commit) to set at least a place holder value for `Status.Version`

File: pkg/operatorstatus/operatorstatus.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+func (o *CVOOperatorStatusHandler) setOperatorStatus(namespace, name string, condtype osapi.ClusterStatusConditionType, status osapi.ConditionStatus, msg string) error {
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		state, err := o.SDKwrapper.Get(name, namespace)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return fmt.Errorf("failed to get cluster operator resource %s/%s: %s", state.ObjectMeta.Namespace, state.ObjectMeta.Name, err)
+			}
+
+			//TODO eventually we will want this off of the commit or something that correlates to versions in the code repo
+			state.Status.Version = "v0.0.1"
```
Comment (bparees) [inline pkg/operatorstatus/operatorstatus.go]: thanks @gabemontero !

File: deploy/03-rbac.yaml [hunk 2]
```diff
@@ -10,6 +10,10 @@ rules:
   - "*"
   verbs:
   - "*"
+- apiGroups:
+  - "operatorstatus.openshift.io"
+  resoruces:
```
Comment (bparees) [inline deploy/03-rbac.yaml]: should be "resources"

File: deploy/03-rbac.yaml [hunk 3]
```diff
@@ -10,6 +10,10 @@ rules:
...
+    - clusteroperators
```
Comment (bparees) [inline deploy/03-rbac.yaml]: indentation level is wrong (which makes this invalid yaml)

Comment (bparees) [reply deploy/03-rbac.yaml]: also you must specify a verbs stanza also.

Comment (bparees) [reply deploy/03-rbac.yaml]: here's a full version of the corrected file:
https://gist.github.com/bparees/8d671e02adfa20c741dbfc30145ab6fc

Comment (gabemontero) [reply deploy/03-rbac.yaml]: updated with gist contents and pushed ... thanks

**Comment by bparees** [comment]
replaced by https://github.com/openshift/cluster-samples-operator/pull/36

**Pull request closed without merging**